src/cares_wrap.cc

```diff
@@ -69,6 +69,21 @@ using v8::Value;
 
 namespace {
 
+inline uint16_t cares_get_16bit(const unsigned char* p) {
+  return static_cast<uint16_t>(static_cast<uint32_t>(0xffff) &
+                               (static_cast<uint32_t>(p[0] << 8U) |
+                               (static_cast<uint32_t>(p[1]))));
+}
+
+inline uint32_t cares_get_32bit(const unsigned char* p) {
+  return static_cast<uint32_t>(static_cast<uint32_t>(p[0] << 24U) |
+                               static_cast<uint32_t>(p[1] << 16U) |
+                               static_cast<uint32_t>(p[2] << 8U) |
+                               static_cast<uint32_t>(p[3]));
```

Comment:
Same you, at least you don’t need the `static_cast<uint32>()` of the return value